### PR TITLE
Added option to update labels & description in project

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added option to update labels & description in project 
- Refactor(Moved) project flag for update & create
- Added Option to update project through file
- Deprecated `--activateProject` `--archiveProject` to `--activate --archive`
-  Change the  file formate for creating a  project from 

Test:
```bash
21:56:15 ➜ flytectl go run main.go create project  --id data --name data --description "Team that manage Data" --labels "team=eta,managedby=flyte"
project Created successfully
21:56:22 ➜ flytectl go run main.go get projects data -oyaml
description: Team that manage Data
domains:
- id: development
  name: development
- id: staging
  name: staging
- id: production
  name: production
id: data
labels:
  values:
    managedby: flyte
    team: eta
name: data

21:56:38 ➜ flytectl go run main.go update project  --id data --name datadata --description "Team that manage Data Platform" --labels "team=data,managedby=flyte"
Project data updated

21:57:40 ➜ flytectl go run main.go get projects datadata -oyaml
description: Team that manage Data Platform
domains:
- id: development
  name: development
- id: staging
  name: staging
- id: production
  name: production
id: data
labels:
  values:
    managedby: flyte
    team: data
name: datadata


21:58:13 ➜ flytectl go run main.go get projects
 --------------- --------------- --------------------------------
| ID (4)        | NAME          | DESCRIPTION                    |
 --------------- --------------- --------------------------------
| data          | datadata      | Team that manage Data Platform |
 --------------- --------------- --------------------------------
| flyteexamples | flyteexamples | flyteexamples description      |
 --------------- --------------- --------------------------------
| flytetester   | flytetester   | flytetester description        |
 --------------- --------------- --------------------------------
| flytesnacks   | flytesnacks   | flytesnacks description        |
 --------------- --------------- --------------------------------

22:00:03 ➜ flytectl go run main.go create project  --file project.yaml
project Created successfully
22:00:50 ➜ flytectl go run main.go get projects mlops -oyaml
description: flytectl feature test labels
domains:
- id: development
  name: development
- id: staging
  name: staging
- id: production
  name: production
id: mlops
labels:
  values:
    managed-by: flyte
    team: mlops
name: mlops


22:01:17 ➜ flytectl go run main.go update project  --file="project.yaml"
Project mlops updated
22:01:57 ➜ flytectl go run main.go get projects ops -oyaml
description: flytectl feature test labels for ops
domains:
- id: development
  name: development
- id: staging
  name: staging
- id: production
  name: production
id: mlops
labels:
  values:
    managed-by: ops
    team: mlops
name: ops


22:02:10 ➜ flytectl go run main.go update project  --file="project.yaml" --archiveProject
Project mlops updated
22:02:43 ➜ flytectl go run main.go get projects
 --------------- --------------- --------------------------------
| ID (4)        | NAME          | DESCRIPTION                    |
 --------------- --------------- --------------------------------
| data          | datadata      | Team that manage Data Platform |
 --------------- --------------- --------------------------------
| flyteexamples | flyteexamples | flyteexamples description      |
 --------------- --------------- --------------------------------
| flytetester   | flytetester   | flytetester description        |
 --------------- --------------- --------------------------------
| flytesnacks   | flytesnacks   | flytesnacks description        |
 --------------- --------------- --------------------------------


21:41:11 ➜ flytectl go run main.go update  project  --id eta  --archiveProject  --activateProject            
Error: Invalid state passed. Specify either activate or archive

exit status 1

22:04:26 ➜ flytectl go run main.go update project  --file="project.yaml" --activateProject
Project mlops updated

22:05:10 ➜ flytectl go run main.go get projects                                           
 --------------- --------------- -------------------------------------- 
| ID (5)        | NAME          | DESCRIPTION                          |
 --------------- --------------- -------------------------------------- 
| mlops         | ops           | flytectl feature test labels for ops |
 --------------- --------------- -------------------------------------- 
| data          | datadata      | Team that manage Data Platform       |
 --------------- --------------- -------------------------------------- 
| flyteexamples | flyteexamples | flyteexamples description            |
 --------------- --------------- -------------------------------------- 
| flytetester   | flytetester   | flytetester description              |
 --------------- --------------- -------------------------------------- 
| flytesnacks   | flytesnacks   | flytesnacks description              |
 --------------- --------------- -------------------------------------- 



19:33:55 ➜ flytectl go run main.go get project lyft -oyaml               
description: spotify feature test labels
domains:
- id: development
  name: development
- id: staging
  name: staging
- id: production
  name: production
id: lyft
labels:
  values:
    managed-by: spotify
    team: spotify
name: spotify

19:34:20 ➜ flytectl go run main.go get project lyft -oyaml > project.yaml
19:34:29 ➜ flytectl go run main.go update project --file project.yaml    
Project lyft updated
19:35:12 ➜ flytectl go run main.go get project lyft -oyaml               
description: Intel feature test labels
domains:
- id: development
  name: development
- id: staging
  name: staging
- id: production
  name: production
id: lyft
labels:
  values:
    managed-by: intel
    team: intel
name: intel

```
## Type
- [x] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [x] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1873
https://github.com/flyteorg/flyte/issues/1620

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by evalsocket** [comment]
Do we also need name update

File: cmd/update/project.go [hunk 1]
```diff
@@ -5,36 +5,30 @@ import (
...
-
 const (
 	projectShort = "Update project resources"
 	projectLong  = `
 Updates the project according to the flags passed. Allows you to archive or activate a project.
 Activate project flytesnacks:
 ::
 
- flytectl update project -p flytesnacks --activateProject
+ flytectl update project --id flytesnacks --activateProject
```
Comment (evalsocket) [inline cmd/update/project.go:25]: @pmahindrakar-oss It will break the backward compatibility

File: cmd/update/project.go [hunk 2]
```diff
@@ -48,45 +42,106 @@ Incorrect usage when passing unknown-project:
...
+		err = yaml.Unmarshal(yamlFile, &projectSpec)
+		if err != nil {
+			return err
+		}
+	} else {
+		projectSpec.ID = project.DefaultProjectConfig.ID
+		projectSpec.Description = project.DefaultProjectConfig.Description
+		projectSpec.Name = project.DefaultProjectConfig.Name
+		projectSpec.Labels = project.DefaultProjectConfig.Labels
+	}
```
Comment (pmahindrakar-oss) [inline cmd/update/project.go:103]: Move this to getProjectSpecFromConfig

File: cmd/config/subcommand/project/project_config.go [hunk 3]
```diff
@@ -11,7 +11,33 @@ var (
...
 type Config struct {
 	Filter filters.Filters `json:"filter" pflag:","`
 }
+
+type Definition struct {
+	ID          string            `yaml:"id"`
+	Name        string            `yaml:"name"`
+	Description string            `yaml:"description"`
+	Labels      map[string]string `yaml:"labels"`
+}
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/project/project_config.go:24]: Any reason we have Definition and same properties part of the ConfigProject.
Instead the definition can be embedded inside ConfigProject to avoid duplication of attributes

File: cmd/update/project.go [hunk 4]
```diff
@@ -5,36 +5,30 @@ import (
...
-
 const (
 	projectShort = "Update project resources"
 	projectLong  = `
 Updates the project according to the flags passed. Allows you to archive or activate a project.
 Activate project flytesnacks:
 ::
 
- flytectl update project -p flytesnacks --activateProject
+ flytectl update project --id flytesnacks --activateProject
```
Comment (pmahindrakar-oss) [inline cmd/update/project.go:26]: This breaks the older functionality. We should keep this backward compatible since incase this is used by teams in there automation already as it would break for them after an upgrade

File: cmd/update/project.go [hunk 5]
```diff
@@ -48,45 +42,106 @@ Incorrect usage when passing unknown-project:
...
+	archiveProject := project.DefaultProjectConfig.ArchiveProject
+	if activateProject || archiveProject {
+		if activateProject == archiveProject {
+			return fmt.Errorf(clierrors.ErrInvalidStateUpdate)
+		}
+		projectDefinition.State = admin.Project_ACTIVE
+		if archiveProject {
+			projectDefinition.State = admin.Project_ARCHIVED
+		}
+	}
```
Comment (pmahindrakar-oss) [inline cmd/update/project.go:134]: Move this to validate function along with lines from 105 :106

File: cmd/update/project.go (hunk 2 above)
Comment (evalsocket) [reply cmd/update/project.go:103]: Created a common method for reading the config in util

File: cmd/update/project.go (hunk 4 above)
Comment (evalsocket) [reply cmd/update/project.go]: reverted the change

File: cmd/create/project.go [hunk 6]
```diff
@@ -31,65 +31,34 @@ Create a project by definition file. Note: The name shouldn't contain any whites
     id: "project-unique-id"
     name: "Name"
     labels:
-     app: flyte
+       values:
+         app: flyte
     description: "Some description for the project"
```
Comment (evalsocket) [inline cmd/create/project.go:36]: @EngHabu This change is not backward compatibly, Is it ok to change the file structure

Comment (evalsocket) [reply cmd/create/project.go:36]: It's ok to break the backward compatibility

**Review by EngHabu — DISMISSED** [review]
The change looks fine to me but it is considered a breaking change because we are removing flags that would work today... Ideally we only deprecate these flags and delete them in a future update... otherwise we should include breaking change notice in the release notes

File: pkg/util/util.go [hunk 7]
```diff
@@ -88,3 +93,27 @@ func SendRequest(method, url string, option io.Reader) (*http.Response, error) {
...
+	}
+
+	projectSpec.Id = id
+	projectSpec.Name = flags.Name
+	projectSpec.Description = flags.Description
+	projectSpec.Labels = &admin.Labels{
+		Values: flags.Labels,
+	}
+	return &projectSpec, nil
+}
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:119]: Instead of adding this to the util, this can be member function for project.ConfigProjec

(p *project.ConfigProject) func  GetProjectSpec() {
    // And this can have logic for all the flag reads / flag  read from file
}

File: cmd/update/project.go [hunk 8]
```diff
@@ -4,89 +4,147 @@ import (
...
+		if activate == archive {
+			return spec, fmt.Errorf(clierrors.ErrInvalidStateUpdate)
+		}
+		spec.State = admin.Project_ACTIVE
+		if archive {
+			spec.State = admin.Project_ARCHIVED
+		}
+		return spec, nil
+	}
+	return spec, nil
```
Comment (pmahindrakar-oss) [inline cmd/update/project.go:149]: Instead of passing in the spec and also having to return the same object ,
We can have this (f *project.ConfigProject) func mapToAdminState() (Project_ProjectState, error) 
and have the spec assignment be done in the consumer call

**Review by pmahindrakar-oss — APPROVED** [review]

File: cmd/config/subcommand/project/project_config.go [hunk 9]
```diff
@@ -11,7 +18,78 @@ var (
...
+	projectSpec.Name = c.Name
+	projectSpec.Description = c.Description
+	projectSpec.Labels = &admin.Labels{
+		Values: c.Labels,
+	}
+	return &projectSpec, nil
+}
+
+//MapToAdminState return project spec from a file/flags
+func (c *ConfigProject) MapToAdminState(spec *admin.Project) (*admin.Project, error) {
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/project/project_config.go:73]: Just return state instead of project and you dont need to send the spec input, 
just use the returned value of the state in the consuming function to assign the returned state

File: cmd/update/project.go [hunk 10]
```diff
@@ -4,89 +4,122 @@ import (
...
+	if projectSpec.Id == "" {
+		return fmt.Errorf(clierrors.ErrProjectNotPassed)
 	}
-	projectState := admin.Project_ACTIVE
-	if archiveProject {
-		projectState = admin.Project_ARCHIVED
+
+	projectDefinition := &admin.Project{
+		Id: projectSpec.Id,
+	}
```
Comment (pmahindrakar-oss) [inline cmd/update/project.go:98]: Why projectDefinition and projectSpec both

**Review by pmahindrakar-oss — APPROVED** [review]

File: cmd/config/subcommand/project/project_config.go [hunk 11]
```diff
@@ -11,7 +18,75 @@ var (
...
+	projectSpec.Name = c.Name
+	projectSpec.Description = c.Description
+	projectSpec.Labels = &admin.Labels{
+		Values: c.Labels,
+	}
+	return &projectSpec, nil
+}
+
+//MapToAdminState return project spec from a file/flags
+func (c *ConfigProject) MapToAdminState() (admin.Project_ProjectState, error) {
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/project/project_config.go:72]: Missing coverage for this function

**Review by pmahindrakar-oss — APPROVED** [review]

File: cmd/config/subcommand/project/project_config.go (hunk 11 above)
Comment (evalsocket) [reply cmd/config/subcommand/project/project_config.go:72]: It's auto merge, I will fix it in follow up PR
